### PR TITLE
Fix Build Options of `GCPImageBuilder`

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
Fixed build options format for `GCPImageBuilder` and also added stack validation before image building to ensure the image builder is in a compatible stack before trying to build.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/gcp/image_builders/gcp_image_builder.py [hunk 1]
```diff
@@ -130,6 +130,9 @@ def build(
             build_context=build_context,
             parent_path_directory_name="cloud-build-contexts",
         )
+        docker_build_options = {
+            "--" + key: value for key, value in docker_build_options.items()
```
Comment (schustmi) [inline src/zenml/integrations/gcp/image_builders/gcp_image_builder.py:134]: Does this actually work? When I try running a local `docker build --rm True ...` this fails for me 🤔

While checking the code, I also saw that in `_configure_cloud_build` we create a list `docker_build_args` which converts the build options dict into a list, but that seems to be completely unused?

Comment (fa9r) [reply src/zenml/integrations/gcp/image_builders/gcp_image_builder.py:134]: I don't know why but yes https://console.cloud.google.com/cloud-build/builds;region=global/7f5ab3a5-2dd7-467e-aba4-200070db7639?project=zenml-core 😅

Comment (fa9r) [reply src/zenml/integrations/gcp/image_builders/gcp_image_builder.py:134]: > While checking the code, I also saw that in _configure_cloud_build we create a list docker_build_args which converts the build options dict into a list, but that seems to be completely unused?

Hahaha yes it is never used, instead it uses `*build_options`, which only passes in the keys, that's why my "fix" above made it just receive `["--pull", "--rm"]` without the `True` values 😄

File: src/zenml/integrations/gcp/image_builders/gcp_image_builder.py [hunk 2]
```diff
@@ -177,9 +177,12 @@ def _configure_cloud_build(
         # Convert the docker_build_options dictionary to a list of strings
         docker_build_args = []
         for key, value in build_options.items():
+            option = f"--{key}"
             if isinstance(value, list):
                 for val in value:
-                    docker_build_args.extend([key, val])
+                    docker_build_args.extend([option, val])
+            elif value is None or isinstance(value, bool):
+                docker_build_args.append(option)
```
Comment (schustmi) [inline src/zenml/integrations/gcp/image_builders/gcp_image_builder.py:185]: Now passing `rm=False` as an option would be translated to `--rm` right?

(I don't have any idea how to solve this in a nice way btw, seems impossible)

File: src/zenml/integrations/gcp/image_builders/gcp_image_builder.py [hunk 3]
```diff
@@ -177,11 +177,14 @@ def _configure_cloud_build(
         # Convert the docker_build_options dictionary to a list of strings
         docker_build_args = []
         for key, value in build_options.items():
+            option = f"--{key}"
```
Comment (schustmi) [inline src/zenml/integrations/gcp/image_builders/gcp_image_builder.py:180]: So the current implementation now works for the two args that we currently provide which is good, I'm pretty sure it's not generic and will break in some other cases.

I guess we just go with it for now? I don't have any idea which works generically other than not passing those arguments here and instead have them configure it as something separate in the `DockerSettings`.

Maybe something like `DockerSettings.image_builder_build_args` or something, but maybe not necessary for now.

File: src/zenml/new/pipelines/pipeline.py [hunk 4]
```diff
@@ -635,6 +635,7 @@ def _run(
                 )
 
             stack = Client().active_stack
+            stack.validate()
```
Comment (schustmi) [inline src/zenml/new/pipelines/pipeline.py:638]: This validation in already happening in `stack.prepare_pipeline_deployment` later in this method. I guess you want it here so we make sure the image builder doesn't break right? Can we somehow verify that at stack registration/update? I guess not as it won't be called when registering from the dashboard at least?

**Review by schustmi — CHANGES_REQUESTED** [review]

File: src/zenml/new/pipelines/pipeline.py (hunk 4 above)
Comment (fa9r) [reply src/zenml/new/pipelines/pipeline.py:638]: Yeah this is technically redundant but I added the call here to ensure we'll always validate before trying to run any pipelines on a stack

**Pull request merged**